### DISCUSSION TITLE
Split jellyfin

### DISCUSSION THREAD

**Discussion by pums974** [jellyfin/jellyfin#7307] [General]
Thanks for this project, It is really great.

The recent development of an alternative to plex/emby/jellyfin ([midarr](https://github.com/midarrlabs/midarr-server)) made me think that it might be relevant to split those projects into two.

1. jellyfin-manager:  the media manager with the database and static part of the social experience (user ratings for example)
2. jellyfin-player: the transcoder server with a web player and the dynamic part of the social experience (online and currently watching for example)

jellyfin-manager would:

* manage posters, metadata, collections, ...
* store the static part of the social experience: watched status / user ratings ... (synced with trakt/imdb/...)
* offer an stable API to add/remove content (for sonarr/radarr/...) in order optimize content discovery
* offer an stable API to request any information about the content (mainly for Plex/jellyfin-player/Emby/kodi/...)
* **maybe** manage media requests (like [ombi](https://github.com/tidusjar/Ombi) or [overserr](https://github.com/sct/overseerr))
* **maybe** display the posters (like [posterr](https://github.com/petersem/poster))

jellyfin-player would be focused on the watching experience which is difficult enough.

This would be more in the KISS design principle, and allows one to

* switch more easily from one player/platform to another
* have the database on one server and the player on another
* simplify the design of each app

(this issue has first been raised on a [midarr issue](https://github.com/midarrlabs/midarr-server/issues/10) but is also relevant for jellyfin

**Comment by oddstr13** [comment]
Jellyfin is _already_ split into multiple parts.

There is jellyfin-server (this repo), and multiple clients (the default one is jellyfin-web, and comes bundled with the server in many cases) https://jellyfin.org/clients/

The server supports plugins, which means that for example individual metadata providers (such as trakt, imdb, tvdb, ...) can be managed and released separately from the server itself.
The server is responsible for managing the media files and their metadata. It provides an API for clients to interface with, and provides transcoding if needed.

The client(s) allow users to log in, browse the library, view the content and track playback progress, all using the APIs provided by the server.

Adding content is done by placing the files in the media directory, and making Jellyfin scan for new content to discover it (either on a timer, or via inotify).
Media requests is _out of scope_ for the project (you know, do one thing and do it well…)
Displaying posters would be up to metadata providers to get the images, and clients to display them (a new image category may be needed for posters).

**Reply by pums974** [reply]
Thanks for the answer. But maybe I wasn't clear. What I mean is precisely to split the transcoder from the media manager. For example, the media manager could be on a NAS without any gpu or hardware acceleration whatsoever, and the transcoder on another system.

I agree that requests a poster were bad ideas (even with emphasized maybes)

**Reply by oddstr13** [reply]
Utilizing remote systems is technically possible right now by using this ffmpeg wrapper; https://github.com/joshuaboniface/rffmpeg
I have not tried the tool myself.
